src/python/pants/base/revision.py

```diff
@@ -93,6 +93,18 @@ def components(self):
   def _is_valid_operand(self, other):
     return hasattr(other, '_components')
 
+  def _fill_value_if_missing(self, ours, theirs):
+    if theirs is None:
+      return ours, type(ours)()
```

Comment:
`type(v)()` is a way to get the zero value for that type. It's equivalent to, for example, calling `str()`.

I played around with a couple other implementations for this function, including a dictionary that associates the type with corresponding default value. This seemed to be the cleanest and most generic approach, but I do recognize `type(ours)()` may look cryptic. Happy to add a comment or go back to that dictionary implementation if you all prefer.

Reply:
A comment would be good. I've never seen this pattern, and would be confused by it.